packetery/src/Packetery/class-message-manager.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Class Message_Manager
+ *
+ * @package Packetery
+ */
+
+declare( strict_types=1 );
+
+
+namespace Packetery;
+
+/**
+ * Class Message_Manager
+ *
+ * @package Packetery
+ */
+class Message_Manager {
+
+	/**
+	 * Latte engine.
+	 *
+	 * @var \Latte\Engine
+	 */
+	private $latte_engine;
+
+	/**
+	 * Messages to be displayed.
+	 *
+	 * @var array
+	 */
+	private $messages;
+
+	/**
+	 * Message_Manager constructor.
+	 *
+	 * @param \Latte\Engine $latte_engine Latte engine.
+	 */
+	public function __construct( \Latte\Engine $latte_engine ) {
+		$messages = \get_transient( 'packetery_message_manager_messages' );
+		if ( ! $messages ) {
+			$messages = array();
+		}
+
+		$this->messages     = $messages;
+		$this->latte_engine = $latte_engine;
+	}
+
+	/**
+	 * Flashes messages to end user.
+	 *
+	 * @param string $message Text.
+	 * @param string $type Type of message.
+	 */
+	private function flash_message( string $message, string $type = 'success' ): void {
+		$message          = array(
+			'type'    => $type,
+			'message' => $message,
+		);
+		$this->messages[] = $message;
+
+		\set_transient( 'packetery_message_manager_messages', $this->messages, 120 );
+	}
+
+	/**
+	 * Shows error message to end user.
+	 *
+	 * @param string $message Message.
+	 */
+	public function flash_error_message( string $message ): void {
```

Comment:
Není tahle metoda zbytečná?

Reply:
Metoda se normálně používá, takže jí jako zbytečnou nevnímám. Zjednodušuje používání ->flash_message(), protože se nemusíš zamýšlet nad typem zprávy.

Reply:
i já se přikláním k tomu,  že zatím se jedná o zbytečnou metodu.  Metoda flash_message má druhý parametr typ a nevidím důvod, proč tento parametr nevyužít. Místo flash_error_message lze tak volat přímo flash_message a přidat druhý parametr,  typ message může být jako konstanta.  Důvod k použití flash_error_message by byl, pokud by metoda dělala i něco jiného, než čisté volání flash_message s parametrem.  Nechceme mít pro každý typ message vlastní metodu.